urld0f681c558_farom57_AstroidDriver_tgz-pJ8-laazotea_indi_server_examples_MinimumINDIServerJ8/src/laazotea/indi/driver/INDISwitchElement.java

```diff
@@ -0,0 +1,145 @@
+ *  of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ * 
+ *  You should have received a copy of the GNU General Public License
+ *  along with INDI for Java Driver.  If not, see 
+ *  <http://www.gnu.org/licenses/>.
+ */
+package laazotea.indi.driver;
+
+import laazotea.indi.Constants;
+import laazotea.indi.Constants.SwitchRules;
+import laazotea.indi.Constants.SwitchStatus;
+import org.w3c.dom.Element;
+
+/**
+ * A class representing a INDI Switch Element.
+ *
+ * @author S. Alonso (Zerjillo) [zerjioi at ugr.es]
+ * @version 1.11, March 26, 2012
+ */
+public class INDISwitchElement extends INDIElement {
+
+  /**
+   * Current Status value for this Switch Element.
+   */
+  private SwitchStatus status;
+
+  /**
+   * Constructs an instance of a
+   * <code>INDISwitchElement</code> with a
+   * <code>name</code>, a
+   * <code>label</code> and its initial
+   * <code>status</code>.
+   *
+   * @param property The Property to which this Element belongs.
+   * @param name The name of the Element.
+   * @param label The label of the Element.
+   * @param status The initial status of the Element
+   * @throws IllegalArgumentException
+   */
+  public INDISwitchElement(INDISwitchProperty property, String name, String label, SwitchStatus status) throws IllegalArgumentException {
+    super(property, name, label);
+
+    this.status = status;
+  }
+
+  /**
+   * Constructs an instance of a
+   * <code>INDISwitchElement</code> with a
+   * <code>name</code>, a
+   * <code>label</code> and its initial
+   * <code>status</code>. The label of the Element will be a copy of the
+   * <code>name</code>.
+   *
+   * @param property The Property to which this Element belongs.
+   * @param name The name of the Element.
+   * @param status The initial state of the Element.
+   * @throws IllegalArgumentException
+   */
+  public INDISwitchElement(INDISwitchProperty property, String name, SwitchStatus status) throws IllegalArgumentException {
+    super(property, name);
+
+    this.status = status;
+  }
+
+  @Override
+  public INDISwitchProperty getProperty() {
+    return (INDISwitchProperty)super.getProperty();
+  }
+
+  @Override
+  public SwitchStatus getValue() {
+    return status;
+  }
+
+  /**
+   * Sets the Element value to a new value. This method ensures that if the
+   * Switch Property rule is
+   * <code>AT_MOST_ONE</code> or
+   * <code>ONE_OF_MANY</code> and the new value is
+   * <code>ON</code> the other Switch Elements of the property are turn to
+   * <code>OFF</code>.
+   *
+   * @param newValue The new value.
+   * @throws IllegalArgumentException If the <code>newValue</code> is not a
+   * valid <code>SwitchStatus</code>.
+   */
+  @Override
+  public void setValue(Object newValue) throws IllegalArgumentException {
+    SwitchStatus ss = null;
+    try {
+      ss = (SwitchStatus)newValue;
+    } catch (ClassCastException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)